File: LayoutTests/http/tests/multipart/policy-ignore-crash.php
<?php
header('Content-type: multipart/x-mixed-replace;boundary=asdf');
?>--asdf
Content-type: text/html

<p>This test passes if it does not crash.</p>
<script>
if (window.layoutTestController)
    layoutTestController.dumpAsText();
</script>

<?php
# Add some padding because CFNetwork merges small multipart segments together.
echo str_pad('', 5000);

ob_flush();
flush();
?>
--asdf
Content-type: text/random

This chunk doesn't have a content type, which can cause the policy
for this load to be ignored. This causes the request to be canceled.

<?php
# Add some padding because CFNetwork merges small multipart segments together.
echo str_pad('', 5000);
?>

--asdf--

File: LayoutTests/http/tests/multipart/policy-ignore-crash.py
#!/usr/bin/env python3

import sys

sys.stdout.buffer.write(
    'Content-Type: multipart/x-mixed-replace;boundary=asdf\r\n\r\n'
    '--asdf\n'
    'Content-type: text/html\n'
    '\n'
    '<p>This test passes if it does not crash.</p>\n'
    '<script>\n'
    'if (window.testRunner)\n'
    '    testRunner.dumpAsText();\n'
    '</script>\n'
    '{}\r\n'.format(' ' * 5000).encode()
)

sys.stdout.flush()

sys.stdout.buffer.write(
    '--asdf\n'
    'Content-type: text/rtf\n'
    '\n'
    'This chunk has an unsupported text mime type, which can cause the policy\n'
    'for this load to be ignored. This causes the request to be canceled.\n'
    '\n'
    '{}\r\n'
    '--asdf--\n'.format(' ' * 5000).encode()
)
